### DISCUSSION TITLE
Custom Tokenization messing up NER

### DISCUSSION THREAD

**Discussion by farrandi** [explosion/spaCy#9168] [Help: Coding & Implementations] [answered]
So I added a custom tokenizer with this extra line for infixes:
```
r"(?<=[{a}])(?=[0-9])".format(a=ALPHA),             # to separate something123
r"(?<=[0-9])(?=[{a}])".format(a=ALPHA),             # to separate 3mmHg
```

So I tried 2 different nlp (one with the custom and one without), and they give different results.
Note that I added these terms as rulers and not a trained model.
This the the code used for both, only the nlp object differs.
```
text = "RVd A4C ... 3mmHg"
doc = nlp(text)
spacy.displacy.render(doc, style = "ent", jupyter = True)
for d in doc:
    print(d)
```

1. Here is the output of the default tokenizer:
![image](https://user-images.githubusercontent.com/54917098/132398211-752ac1b9-769f-41c4-8095-5e3537080930.png)

2. Here is the output of the custom tokenizer:
![image](https://user-images.githubusercontent.com/54917098/132398255-38323caa-3bfd-4eed-8e8a-2a444bf15c4b.png)

Is there a reason the NER does not recognize "RVd A4C" even though it was added as a ruler-based NER to the model.

**Comment by adrianeboyd** [comment, marked as answer]
It depends on how you created the entity ruler patterns, which you haven't described above.

In general, be sure that you make all your tokenizer modifications before adding any entity ruler phrase patterns, and if you add token-based patterns, they need to follow the modified tokenization (just as with the `Matcher`).

**Reply by farrandi** [reply]
Oh I simply added the tokenizer before adding the ruler patterns and now it works! thanks!
